stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -1417,6 +1418,26 @@ for (fname, elty) in ((:dger_,:Float64),
     end
 end
 
+### geru
+
+"""
+    geru!(alpha, x, y, A)
+
+Performs an unconjugated rank-1 update of the matrix `A` with vectors `x` and `y` as `alpha*x*transpose(y) + A`.
+"""
+function geru!(α::$elty, x::AbstractVector{$elty}, y::AbstractVector{$elty}, A::AbstractMatrix{$elty})
+    if isreal(α)
```

Comment:
It seems there should be errors if one input is complex and not the other (rather than silently doing nothing)

Reply:
These are strange conditions anyway. What is `$elty` in the first place? Second, all (el)types seem to be (intended to be) restricted to be the same. Third, the distinction in the function body could be done with multiple dispatch:

```julia
geru!(α::Real, x::AbstractVector{<:Real}, y::AbstractVector{<:Real}, A::AbstractMatrix{<:Real}) =
    BLAS.ger!(α, x, y, A)
geru!(α::Complex, x::AbstractVector{<:Complex}, y::AbstractVector{<:Complex}, A::AbstractMatrix{<:Complex}) =
    BLAS.zgeru!(α, x, y, A)
```

or perhaps `Real/Complex` replaced by `BlasReal/BlasComplex`. In the complex version, one could allow `α::BlasFloat` and call `complex(α)` in the `BLAS.zgeru!` call.